src/azure-cli/azure/cli/command_modules/vm/_template_builder.py

```diff
@@ -847,6 +851,22 @@ def build_vmss_resource(cmd, name, naming_prefix, location, tags, overprovision,
     }
     if zones:
         vmss['zones'] = zones
+    # vmss without vm profile
+    if empty:
+        if platform_fault_domain_count is None:
+            platform_fault_domain_count = 2
```

Comment:
Why is it defaulted to 2? Can this not be passed down as null?

Reply:
When I use null,
Deployment failed. Correlation ID: e7f42bc9-2d4b-4310-90cd-8291da117c5d. {
  "error": {
    "code": "InvalidParameter",
    "message": "Required parameter 'platformFaultDomainCount' is missing (null).",
    "target": "platformFaultDomainCount"
  }
}

Reply:
Oh My bad. We do need this one; but not sure defaulting is the right thing to do v/s making clients always enter that value.